.github/workflows/cd.yml

```diff
@@ -0,0 +1,239 @@
+      with:
+        name: SolveSpace.dmg
+        path: build/bin/SolveSpace.dmg
+        
+  deploy-snap-amd64:
+    needs: [test-ubuntu, test-windows, test-macos]
+    name: Deploy AMD64 Snap
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set Up Source
+      run: rsync --filter=":- .gitignore" -r ./ pkg/snap/solvespace-snap-src
+    - name: Build Snap
+      id: build
+      uses: diddlesnaps/snapcraft-multiarch-action@v1
+      with:
+        path: pkg/snap
+    - name: Upload & Release to Edge
+      if: github.event_name == 'push'
+      uses: snapcore/action-publish@v1
+      with:
+        store_login: ${{ secrets.SNAPSTORE_LOGIN }}
+        snap: ${{ steps.build.outputs.snap }}
+        release: edge
+    - name: Upload & Release to Beta + Edge
+      if: github.event_name == 'release'
+      uses: snapcore/action-publish@v1
+      with:
+        store_login: ${{ secrets.SNAPSTORE_LOGIN }}
+        snap: ${{ steps.build.outputs.snap }}
+        release: edge,beta
+    
+  deploy-snap-arm64:
+    needs: [test-ubuntu, test-windows, test-macos]
+    name: Deploy ARM64 Snap
+    runs-on: ubuntu-latest
+    steps:
+    - uses: docker/setup-qemu-action@v1
+    - uses: actions/checkout@v2
+    - name: Set Up Source
+      run: rsync --filter=":- .gitignore" -r ./ pkg/snap/solvespace-snap-src
+    - name: Build Snap
+      id: build
+      uses: diddlesnaps/snapcraft-multiarch-action@v1
+      with:
+        path: pkg/snap
+        architecture: arm64
+    - name: Upload & Release to Edge
+      if: github.event_name == 'push'
+      uses: snapcore/action-publish@v1
+      with:
+        store_login: ${{ secrets.SNAPSTORE_LOGIN }}
+        snap: ${{ steps.build.outputs.snap }}
+        release: edge
+    - name: Upload & Release to Beta + Edge
+      if: github.event_name == 'release'
+      uses: snapcore/action-publish@v1
+      with:
+        store_login: ${{ secrets.SNAPSTORE_LOGIN }}
+        snap: ${{ steps.build.outputs.snap }}
+        release: edge,beta
+  
+  update_edge_release:
+    name: Update Edge Release
+    needs: [build-release-windows, build-release-windows-openmp, build-release-macos]
+    if: always() && github.event_name == 'push'
+    runs-on: ubuntu-latest
+    outputs:
+      upload_url: ${{ steps.create_release.outputs.upload_url }}
+    steps:
+    - name: Delete Old Edge Release
+      uses: dev-drprasad/delete-tag-and-release@v0.1.2
+      with:
+        delete_release: true
+        tag_name: edge
+      env:
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+    - name: Create New Edge Release
+      id: create_release
+      uses: actions/create-release@v1
+      env:
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      with:
+        tag_name: edge
+        release_name: Edge
+        prerelease: true
+        body: ${{ github.event.head_commit.message }}
+
+  upload-release-assets:
+    name: Upload Release Assets
+    needs: [update_edge_release]
```

Comment:
In my tests, this inherited the `needs` from `update_edge_release`, even if that job got skipped. But perhaps it would be more clear to expand it to `needs: [build-release-windows, build-release-windows-openmp, build-release-macos update_edge_release]`.

Reply:
Yes I think that makes sense.
Nitpick: I'd also change the `_` to `-` (or the other way around) to make it more consistent with the rest

Reply:
Yeah. Dashes are not legal in property dot notation, but I need to reference this job later: `needs.update_edge_release.outputs.upload_url`. So underscores it is then.